drivers/serial/uart_nrfx_uarte.c

```diff
@@ -1501,6 +1502,7 @@ static void uarte_nrfx_poll_out(const struct device *dev, unsigned char c)
 			}
 
 			irq_unlock(key);
+			Z_SPIN_DELAY(2);
```

Comment:
is this spin delay something specific to the posix?

Reply:
It defines out to nothing for other architectures.